### DISCUSSION TITLE
How to solve the problem that the box disappears and appears too suddenly?

### DISCUSSION THREAD

**Discussion by feipinxiang** [LanKuDot/game_modules#42] [Q&A] [answered]
I want to create a mask for the CircularScrollingList display area to cover up the box disappear and appear.
![image](https://github.com/LanKuDot/game_modules/assets/2398548/f178b0f1-7db8-4d80-af77-43714b16d4d4)
This is a horizontal list with 5 boxes. The pivot of the box is in the center. The green area is the recttransform of the list. The red area is the display area calculated by CircularScrollingList.
![drag left](https://github.com/LanKuDot/game_modules/assets/2398548/b8f19190-8cb7-4e9d-b68f-93561eeeeceb)
If I set the red area as a mask, when I swipe left, box1 will pop out while **still half visible**, and the next one will pop up in the empty space to the right of the red.
How to solve the problem that the box disappears and appears too suddenly?
![image](https://github.com/LanKuDot/game_modules/assets/2398548/826b437b-26b2-4808-9625-ce154a8556c1)

**Comment by feipinxiang** [comment]
Maybe need to create one more box than numOfBoxes and modify BoxTransformController.minPos/maxPos to solve this problem😅

**Reply by LanKuDot** [reply, marked as answer]
Hi, you could add two more boxes and adjust the `BoxDensity` to make the range of list wider.
The formula of calculating the `BoxDensity` is `(number of boxes before added - 1) / (number of boxes after added - 1)`.
In this case, the list originally has 5 boxes, and 2 more boxes is added. Therefore, the `BoxDensity` is (5 - 1) / (7 - 1) = 0.66666

**Reply by feipinxiang** [reply]
Thanks. This solves the problem I'm currently having (looping type list). But through BoxDensity to solve, if the list is linear, and FocusingPosition is top or button, then the first and last box will never be seen.

**Reply by LanKuDot** [reply]
Sorry about that. I thought it will be the problem when I made the version 6.
The workaround is that remove the additional 2 boxes, and use `OnBoxMoved(float positionRatio)` function to control the transparency of the box, which is available in the `master` branch.
Here is the relationship between the position ratio and the boxes:
![圖片](https://github.com/LanKuDot/game_modules/assets/2039072/4998659f-7e2c-4ddf-9b68-318eead25339)

**Reply by feipinxiang** [reply]
Thank you for your reply. This library saved me a lot of time, thanks a lot. At present, I haven't used Linear list in my project.  I just found out that this happens when I test your library. 
I think it might be a better solution to create another box on the other side before the first box moves from 0.83333 to 1.1666. a substitute player⚽. This will make the range from 1 to -1 appear perfectly instead of disappearing suddenly. After all, the interpretation and calculation of 0.8333 is more troublesome than using 1 directly. But this may make your list sorting algorithm more complicated than it already is.

**Reply by LanKuDot** [reply]
Thanks for the recommendation! I would try to fix this issue in the next version.
This asset has been upgraded many times. New features follow new problems 🥲
